client/k8s.go

```diff
@@ -100,43 +94,35 @@ func GetOutofClusterKubeConfig() {
 		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig")
 	}
 	flag.Parse()
-
 	os.Setenv("KUBECONFIG", *kubeconfig)
```

Comment:
TODO(upcoming PRs): This logic should be there in the CLI(maybe in `cobra.Command.preRun` only so that if someone passes the `--kubeconfig, --token, --user` flags or some token, that and **only that** should be put to use, i.e. after restructuring the k8s clients should be intialized **once** preferably at the sub-command level which is passed to other sub-commands which pass it to the fetching logic as implicit arguments.

The current one is good but refining the above idea might help.